src/main/java/com/comphenix/protocol/injector/collection/OutboundPacketListenerSet.java

```diff
@@ -0,0 +1,88 @@
+package com.comphenix.protocol.injector.collection;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+
+import javax.annotation.Nullable;
+
+import com.comphenix.protocol.PacketType;
+import com.comphenix.protocol.ProtocolLibrary;
+import com.comphenix.protocol.concurrent.PacketTypeListenerSet;
+import com.comphenix.protocol.error.ErrorReporter;
+import com.comphenix.protocol.events.ListenerPriority;
+import com.comphenix.protocol.events.ListeningWhitelist;
+import com.comphenix.protocol.events.PacketContainer;
+import com.comphenix.protocol.events.PacketEvent;
+import com.comphenix.protocol.events.PacketListener;
+
+public class OutboundPacketListenerSet extends PacketListenerSet {
+
+	public OutboundPacketListenerSet(PacketTypeListenerSet mainThreadPacketTypes, ErrorReporter errorReporter) {
+		super(mainThreadPacketTypes, errorReporter);
+	}
+
+	@Override
+	protected ListeningWhitelist getListeningWhitelist(PacketListener packetListener) {
+		return packetListener.getSendingWhitelist();
+	}
+
+	@Override
+	public void invoke(PacketEvent event, @Nullable ListenerPriority priorityFilter) {
+		super.invoke(event, priorityFilter);
+
+		if (event.getPacketType() == PacketType.Play.Server.BUNDLE && !event.isCancelled()) {
+			// unpack the bundle and invoke for each packet in the bundle
+			Iterable<PacketContainer> packets = event.getPacket().getPacketBundles().read(0);
+			List<PacketContainer> outPackets = new ArrayList<>();
+			for (PacketContainer subPacket : packets) {
+				if (subPacket == null) {
+					ProtocolLibrary.getPlugin().getLogger().log(Level.WARNING,
+							"Failed to invoke packet event "
+									+ (priorityFilter == null ? "" : ("with priority " + priorityFilter))
+									+ " in bundle because bundle contains null packet: " + packets,
+							new Throwable());
```

Comment:
here and below, i'd prefer a more descriptive exception type than just `Throwable`. 

also, do we know when a bundle would contain a null packet? is that something that the server would do, or would that point to an issue on our end?

Reply:
I copied this from the old implementation without giving it much thought. However, I'm questioning the necessity of logging null packets maybe we should remove them silently. As far as I know, null packets can only be introduced by plugins using ProtocolLib. This could be problematic, as it might result in a NullPointerException within the packet encoder, potentially disconnecting the client.

Reply:
if a null packet is included in the bundle by someone else, we should ignore it. if a plugin introduces one, we should loudly (i.e. log an exception) remove it but avoid causing errors downstream

Reply:
Ok I will ignore external null packets and add a comment about it.